rosjava_helpers/src/main/java/org/ros/helpers/ParameterLoaderNode.java

```diff
@@ -0,0 +1,138 @@
+
+    public static final String NODE_NAME = "parameter_loader";
+    private final List<LoadedResource> params = new ArrayList<>();
+    private Log log;
+
+    /**
+     * Default constructor
+     * @param resources Array of resources with their respective namespace to load.
+     */
+    public ParameterLoaderNode(ArrayList<Resource> resources) {
+        for (Resource r : resources) {
+            addSingleYmlInput(r.inputStream, r.namespace == null ? "" : r.namespace);
+        }
+    }
+
+    private void addSingleYmlInput(InputStream ymlInputStream, String namespace) {
+        this.params.add(new LoadedResource((new Yaml()).load(ymlInputStream), namespace));
+    }
+
+    private void addParams(ParameterTree parameterTree, String namespace, Map<String, Object> params) {
+        for (Map.Entry<String, Object> e : params.entrySet()) {
+            String fullKeyName = namespace + "/" + e.getKey();
+            if (log != null) {
+                log.info("Loading parameter " + fullKeyName + " \nValue = " + e.getValue());
+            }
+
+            if (e.getValue() instanceof String) {
+                parameterTree.set(fullKeyName, (String)e.getValue());
+            } else if (e.getValue() instanceof Integer) {
+                parameterTree.set(fullKeyName, (Integer)e.getValue());
+            } else if (e.getValue() instanceof Double) {
+                parameterTree.set(fullKeyName, (Double)e.getValue());
+            } else if (e.getValue() instanceof Map) {
+                parameterTree.set(fullKeyName, (Map)e.getValue());
+            } else if (e.getValue() instanceof Boolean) {
+                parameterTree.set(fullKeyName, (Boolean)e.getValue());
+            } else if (e.getValue() instanceof List) {
+                parameterTree.set(fullKeyName, (List)e.getValue());
+            } else if (log != null) {
+                log.debug("I don't know what type parameter " + fullKeyName + " is. Value = " + e.getValue());
+                log.debug("Class name is: " + e.getValue().getClass().getName());
+            }
+        }
+    }
+
+    /**
+     * @return the name of the Node that will be used if a name was not
+     * specified in the Node's associated NodeConfiguration.
+     */
+    @Override
+    public GraphName getDefaultNodeName() {
+        return GraphName.of(NODE_NAME);
+    }
+
+    @Override
+    public void onStart(ConnectedNode connectedNode) {
+        if (params != null) {
+            ParameterTree parameterTree = connectedNode.getParameterTree();
+            log = connectedNode.getLog();
+
+            // TODO: For some reason, setting the / param when using a rosjava master doesn't work
+            // It does work fine with an external master, and also setting other params of any type
+            for (LoadedResource r : params) {
+                addParams(parameterTree, r.namespace, r.resource);
+            }
+
+            connectedNode.shutdown();
```

Comment:
it doesn't need to wait here a little bit before shutdown? This doesn't cause any of the configurations to be lost?